WalletWasabi/JsonConverters/CompactSignatureJsonConverter.cs

```diff
@@ -0,0 +1,76 @@
+using NBitcoin;
+using Newtonsoft.Json;
+
+namespace WalletWasabi.JsonConverters;
+
+public class CompactSignatureJsonConverter : JsonConverter<CompactSignature>
+{
+	/// <inheritdoc />
+	public override CompactSignature? ReadJson(JsonReader reader, Type objectType, CompactSignature? existingValue, bool hasExistingValue, JsonSerializer serializer)
+	{
+		var value = reader.Value as string;
+
+		if (string.IsNullOrWhiteSpace(value))
+		{
+			return null;
+		}
+
+		byte[] bytes = Convert.FromBase64String(value);
+		int recoveryId = ComputeRecoveryId(bytes[0]);
```

Comment:
Should we verify the `recoveryId` is valid or is it validated by the `CompactSignature` ctor?

Reply:
So `recoveryId` is not being validated in https://github.com/MetacoSA/NBitcoin/blob/v7.0.6/NBitcoin/CompactSignature.cs. Also no validation is here https://github.com/MetacoSA/NBitcoin/blob/v7.0.6/NBitcoin/Secp256k1/Recovery/SecpRecoverableECDSASignature.cs either.

But then this looks like the validation you talk about is [here](https://github.com/MetacoSA/NBitcoin/commit/48341d508ef306dd7cfc1f50539acbd3b847ba23#diff-923260e901978d29eb3b7543ea7386c678f02e05a108838d39fe8859ff4acacdL455-L456). I added:

```cs
		if (!SecpRecoverableECDSASignature.TryCreateFromCompact(compactSignature.Signature, compactSignature.RecoveryId, out _))
		{
			throw new InvalidOperationException("Compact signature is not valid.");
		}
```

but it does not validate much imho.